Adds functionality to strength and stimmed

trollbreeder opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
This adds functionality to strength and stimmed, so they're not just craftgenes.
Strength allows you to carry mobs and lockers without being slowed down. Very useful for would-be paramedics, as long as you stop the bleeding. Nyoom.
Stimmed is what i hope will become used often, maybe even a must-have like space adap. It removes toxic reagents slowly but consistently from the user. Dap on the chemist as you shrug off their chloral hydrate cyanide nitric acid fluorosulfuric acid lexorin deathmix! Drink sulfonal as a party trick! Etc etc..

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Fun new mutations to play with, no more throwing away these craftgenes after being used to make a better mutation. Now they have actual purpose to be injected into you, which is good.

## Changelog
:cl:
add: The strength mutation now allows dragging heavy objects with ease.
add: The stimmed mutation now slowly purges toxic reagents.
balance: Stimmed and strength have been given instability
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

trollbreeder committed:
Merge pull request #1 from tgstation/master

Updates master from source repo

actions-user committed:
Automatic changelog compile [ci skip]

trollbreeder committed:
Buffs strength and stimmed

Strength now lets you carry lockers and mobs without slowing down, very useful for paramedics, janitors and cargo techs
Stimmed constantly purges toxins from the body, slowly but consistently.

trollbreeder committed:
Revert "Automatic changelog compile [ci skip]"

This reverts commit d3d93059dde950b9d455a38dd2ab53f1595ddd3a.

trollbreeder committed:
fixes indentations

trollbreeder committed:
This documentation needs to be more specific

trollbreeder commented:
hold on a second

trollbreeder committed:
okay NOW this fixes indentation

Pumpkinoe commented:
woah!!! this is cool and good
does strength stop all slowdown from carrying? dragging?

trollbreeder commented:
It stops slowdown from pulling lockers and mobs, since those use drag_slowdown. Carrying is not handled under this, i think.

Pumpkinoe commented:
maybe this will pave the way to making gigantism have functionality too
strength + giantism when?

Pugie commented:
There's a reason these slowdowns exist - because being dragged away by someone murderboning on meth without other people being able to save you is lame.

TiviPlus commented:
Dragfighting is back in town

trollbreeder committed:
nerf to instability

people were complaining, maybe this will help

PKPenguin321 commented on code/datums/mutations/body.dm:
```diff
@@ -224,17 +224,35 @@
+/datum/mutation/human/strong/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	owner.slowed_by_drag = initial(owner.slowed_by_drag)
```
slowed_by_drag should be made a trait for futureproofing

trollbreeder replied:
i'll look into it

PKPenguin321 commented:
imo Strength will not be worth the 25 instability 99% of the time, seeing as if you have it you're half of the way to having hulk and hulk is just straight up better use of your genetic capacity if you want to murderbone. it's nothing to really worry about

necromanceranne commented:
PKP,  only humans can be hulks.

trollbreeder commented:
So, 20 instability instead?

trollbreeder committed:
buffs instability, adds exclamation mark, and refactors slowed_by_drag into a trait

Instability is now 20, from 25 due to remarks about it being "no ones gonna abuse it, hulk is better anyways"
You feel strong! instead of You feel strong.
Refactors slowed_by_drag into a trait.

MimicFaux commented:
A decent instability would be worthwhile to balance against the reason slowdowns were introduced in the firstplace.

trollbreeder commented:
Slowed_by_drag refactored into a trait, and instability is now 20, from 25.

ATH1909 commented:
Can you make strength also make you ignore the slowdown from fireman carrying someone?

trollbreeder commented:
> Can you make strength also make you ignore the slowdown from fireman carrying someone?

I tried coding that, but i tested it and it doesn't seem like it works. The slowdown from fireman carrying is negligible in my testing.

Rohesie commented on code/datums/mutations/body.dm:
```diff
@@ -224,17 +224,35 @@
+/datum/mutation/human/strong/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_NOT_SLOWEDBYDRAG, "genetics") // Don't get slowed by dragging lockers or mobs
```
```suggestion
	owner.add_movespeed_mod_immunities(type, /datum/movespeed_modifier/bulky_drag)
	owner.add_movespeed_mod_immunities(type, /datum/movespeed_modifier/human_carry)
```

trollbreeder replied:
ah, thanks, that's what i was looking for when i was trying to add carry slowdown immunity

Rohesie commented on code/datums/mutations/body.dm:
```diff
@@ -224,17 +224,35 @@
+/datum/mutation/human/strong/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	REMOVE_TRAIT(owner, TRAIT_NOT_SLOWEDBYDRAG, "genetics")
```
```suggestion```suggestion
	owner.remove_movespeed_modifier(type, /datum/movespeed_modifier/bulky_drag)
	owner.remove_movespeed_modifier(type, /datum/movespeed_modifier/human_carry)
```
```

trollbreeder replied:
you scuffed this suggestion

Rohesie commented on code/datums/mutations/body.dm:
```diff
@@ -224,17 +224,35 @@
+	instability = 20
+
+/datum/mutation/human/strong/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
```
```suggestion
	. = ..()
	if(.)
```

trollbreeder committed:
. = ..()

Co-Authored-By: Rohesie <rohesie@gmail.com>

trollbreeder committed:
Movespeed mod immunities

Co-Authored-By: Rohesie <rohesie@gmail.com>

Rohesie commented on code/datums/mutations/body.dm:
```diff
@@ -224,17 +224,36 @@
+
+/datum/mutation/human/strong/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
```
```suggestion
	. = ..()
	if(.)
		return
```

trollbreeder committed:
Movespeed mod immunity part 2

Co-Authored-By: Rohesie <rohesie@gmail.com>

trollbreeder committed:
. = ..() part 2

Co-Authored-By: Rohesie <rohesie@gmail.com>

trollbreeder commented:
add_movespeed_mod_immunities is an unknown proc

trollbreeder committed:
Removes the not slowed by drag trait in favor of movespeed immunities

Removes the not slowed by drag trait in favor of movespeed immunities that were added in commit 4901adc and d05029a

Rohesie commented:
Update your code.

trollbreeder commented:
I'm doing so

trollbreeder commented:
@Cyberboss Help, the Appveyor gods are angry and they say they cannot download GitHub repository contents because it is forbidden

Donpedrito commented:
@pumpkinoe Gigantism increases your odds of tackling successfully

trollbreeder commented:
i have no fucking idea what i'm doing on the github side and i'm thinking of closing this and making a sequel PR

trollbreeder commented:
Remaking this from an updated master. Keep an eye out for the new PR!
